crucible-llvm/src/Lang/Crucible/LLVM/Extension/Syntax.hs

```diff
@@ -227,6 +227,43 @@ data LLVMStmt (f :: CrucibleType -> Type) :: CrucibleType -> Type where
      !(f (LLVMPointerType wptr)) {- Second pointer -} ->
      LLVMStmt f (BVType wptr)
 
+  -- | Debug information
+  LLVM_Debug ::
+    !(LLVM_Dbg f c)              {- Debug variant -} ->
+    LLVMStmt f UnitType
+
+-- | Debug statement variants - these have no semantic meaning
+data LLVM_Dbg f c where
```

Comment:
I wasn't in the loop on the initial design - what is the motivation for having the `c` type parameter exposed? It is existentially quantified away in the `LLVM_Debug` statement, so there is no type-level connection there. All of the uses in this `LLVM_Dbg` type either fix that type or have an associated repr, so you can definitely already recover those types by pattern matching.

Reply:
I did that because it made it fit into the -FC world of typeclasses more neatly. I'd be happy to replace this with something better. I originally didn't have that parameter and then seemed like I didn't have good support for fixing up the TraversableFC instance for LLVM statements without it.

Reply:
Ah I was afraid of that. Fair enough. If it becomes an issue we can figure out what to do with it later